include/swift/AST/Module.h

```diff
@@ -711,6 +711,18 @@ class ModuleDecl
     Bits.ModuleDecl.IsBuiltFromInterface = flag;
   }
 
+  /// Returns true if -experimental-allow-non-resilient-access was passed and
+  /// the module was built from source without internal contents made visible.
+  bool allowNonResilientAccess() const {
+    return Bits.ModuleDecl.AllowNonResilientAccess &&
+          !Bits.ModuleDecl.IsBuiltFromInterface &&
+          !Bits.ModuleDecl.TestingEnabled &&
+          !Bits.ModuleDecl.PrivateImportsEnabled;
```

Comment:
These checks are done at the frontend args converter but added here for extra assurance; maybe this is redundant?